### ISSUE TITLE
RFC: Do Flow types still need to be exported after the TS Conversion?

### ISSUE THREAD

**Issue by mike-marcacci** [graphql/graphql-js#2832]
In [the working group meeting today](https://github.com/graphql/graphql-js-wg/blob/main/agendas/2020-10-28.md), we discussed the path forward for the conversion to TypeScript. Per @IvanGoncharov there had originally been a commitment that this library would continue to export flow types as best as it could. However, this commitment was made with the expectation that the conversion would be finished about a year ago.

Given that so much time has elapsed, it's definitely worth reassessing the value of continuing to export flow types. The maintenance burden of maintaining parallel types is not trivial, as it either adds a substantial barrier to entry for new contributors, or requires maintainers to keep the flow types in sync with each accepted contribution.

But while it may be preferable to drop flow types from a maintenance standpoint, it's completely possible that enough users in the community depend on them that the continued cost is justified. (I actually quite suspect this is the case, but it would be foolish for us not to check.)

So as a question to all stakeholders (that's anybody who uses graphql-js in an app or library): **Do you use the flow types exported by graphql-js?** 

And particularly for those who were party to the original agreement and the folks at Facebook: **Is continued support for flow types still important? Have the dependent projects migrated to TypeScript or otherwise lost the need for these types?**

**Comment by magicmark** [comment]
At Yelp, we use GraphQL pretty heavily and have built a fair amount of tooling (for linting, schema registry management, persisted queries) on top of graphql-js. We also use Flow, so we are importing a lot of its Flow types.

That said, we use lots of projects not written with Flow - consuming types from https://github.com/flow-typed/flow-typed is a more common workflow for us.  (I suspect the occasional running of https://github.com/joarwilk/flowgen could get us 80% the way there)

So tl;dr, to answer your question:

> Do you use the flow types exported by graphql-js?

Yes

(fwiw I agree that TypeScript makes as a direction for graphql-js - I think we probably want to make the same switch soon, just haven't been able to prioritize it yet)

Thanks!

**Comment by IvanGoncharov** [comment]
We disscussed it on April WG and discussed that FB engineers are ok with drop support for Flow types in 16.0.0.
And they freeze their Flow code on last awaible version supporting Flow types.
cc: @mjmahone

**Comment by yaacovCR** [comment]
This can then probably be closed?

**Comment by yaacovCR** [comment]
And unpinned?

**Comment by yaacovCR** [comment]
And unpinned?

**Comment by saihaj** [comment]
@IvanGoncharov can you please unpin this?
